### PR TITLE
[SofaConstraint] Clean timers in LCPConstraintSolver

### PR DESCRIPTION
**Pull request by alxbilger**
Timers have been transformed to `AdvancedTimer`.
`MechanicalGetConstraintInfoVisitor` has now its own file.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

File: modules/SofaConstraint/src/SofaConstraint/LCPConstraintSolver.cpp [hunk 1]
```diff
@@ -44,6 +45,56 @@ using sofa::core::VecId;
 namespace sofa::component::constraintset
 {
 
+class MechanicalGetConstraintInfoVisitor : public simulation::BaseMechanicalVisitor
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/LCPConstraintSolver.cpp]: Any reason to set this Visitor directly into the cpp ? I saw it was in the LCPCSolver header before, could not it be in a separate file (along the other mechavisitor) ?
Or is it only useful for LCPSolver and thats why you set in the cpp ("kind of private" class) ?

Comment (alxbilger) [reply modules/SofaConstraint/src/SofaConstraint/LCPConstraintSolver.cpp]: Yes, this visitor is used only in this LCPSolver. I did not like to see the visitor class in the header file, as it hides the main class LCPSolver. I don't mind putting it in a dedicated file if requested.

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

**Comment by alxbilger** [comment]
SofaConstraint links with SofaImplicitOdeSolver, which links with Sofa.Component.ODESolver.Backward, which links with SofaSimulationCore. Isn't the dependencies transferred in this case?

I launch a full build, just in case.
[ci-build][force-full-build][with-all-tests]

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

**Pull request merged**